lib/matplotlib/axes/_axes.py

```diff
@@ -7234,6 +7241,15 @@ def stairs(self, values, edges=None, *,
                                    fill=fill,
                                    **kwargs)
         self.add_patch(patch)
+        if baseline is None and fill:
+            _api.warn_external(
+                f"Both {baseline=} and {fill=} have been passed. "
+                "Because baseline in None, the Path used to draw the stairs will "
```

Comment:
```suggestion
                "baseline=None is only intended for unfilled stair plots. "
                "Because baseline in None, the Path used to draw the stairs will "
```